libs/application/data-providers/src/providers/PaymentCatalogProvider.ts

```diff
@@ -1,86 +1,34 @@
 import { PaymentCatalogItem } from '@island.is/api/schema'
-import {
-  BasicDataProvider,
-  Application,
-  SuccessfulDataProviderResult,
-  FailedDataProviderResult,
-} from '@island.is/application/core'
+import { BasicDataProvider } from '@island.is/application/core'
 
-const CHARGE_ITEM_CODE = 'AY110'
-
-// TODO: needs refactoring
-const searchCorrectCatalog = (
-  keySearch: string,
-  searchJSON: string,
-): string | null => {
-  if (keySearch == '' || searchJSON == '') {
-    return searchJSON
-  }
-  const resultCatalog = JSON.parse(searchJSON)
-  for (const item in resultCatalog) {
-    if (resultCatalog[item].chargeItemCode == keySearch) {
-      return resultCatalog[item]
-    }
-  }
-  return null
-}
-
-export class PaymentCatalogProvider extends BasicDataProvider {
-  type = 'PaymentCatalogProvider'
-
-  async provide(application: Application): Promise<PaymentCatalogItem[]> {
+export abstract class PaymentCatalogProvider extends BasicDataProvider {
+  async getCatalogForOrganization(
+    organizationId: string,
+  ): Promise<PaymentCatalogItem[] | null> {
     const query = `
-    query PaymentCatalogProvider($input: PaymentCatalogInput!) {
-      paymentCatalog(input: $input) {
-        items {
-          performingOrgID
-          chargeType
-          chargeItemCode
-          chargeItemName
-          priceAmount
+      query PaymentCatalogProvider($input: PaymentCatalogInput!) {
+        paymentCatalog(input: $input) {
+          items {
+            performingOrgID
+            chargeItemCode
+            chargeItemName
+            priceAmount
+          }
         }
       }
-    }
     `
 
     return this.useGraphqlGateway(query, {
-      input: { performingOrganizationID: '6509142520' },
+      input: { performingOrganizationID: organizationId },
     }).then(async (res: Response) => {
-      // TODO: needs refactoring
       const response = await res.json()
+
       if (response.errors) {
+        console.error(response)
```

Comment:
I don't see why we shouldn't be using `@island.is/logging` here, but agreed on throwing stead of returning a rejected promise

Reply:
Actually the logger doesn't work in this context. There's some import issues with node packages in the logger. Probably worth exploring separately.

Reply:
Yeah, we have these issues because the template libs are isomorphic (used in the website as well as on the backend). We've wanted to address this for some time now but it's a big task... :(